### PR TITLE
 feat: Adds the possibility to create VP based on the definition

### PR DESCRIPTION
**Pull request by soluchok**
Adds the possibility to create VP based on the definition.

Signed-off-by: Andrii Soluk <isoluchok@gmail.com>

### CODE REVIEW

File: pkg/doc/presexch/definition.go [hunk 1]
```diff
@@ -166,3 +168,90 @@ func (pd *PresentationDefinition) ValidateSchema() error {
...
+
+// CreateVP creates verifiable presentation.
+func (pd *PresentationDefinition) CreateVP(credentials ...*verifiable.Credential) (*verifiable.Presentation, error) {
+	if err := pd.ValidateSchema(); err != nil {
+		return nil, err
+	}
+
+	if len(pd.SubmissionRequirements) > 0 {
+		// TODO: handle submission requirements
+		return nil, errors.New("not supported yet")
```
Comment (llorllale) [inline pkg/doc/presexch/definition.go:180]:
```suggestion
		return nil, errors.New("submission requirements not supported yet")
```

File: pkg/doc/presexch/definition.go [hunk 2]
```diff
@@ -166,3 +168,90 @@ func (pd *PresentationDefinition) ValidateSchema() error {
...
+	}
+
+	var result []*verifiable.Credential
+
+	for _, descriptor := range pd.InputDescriptors {
+		filtered := filterSchema(descriptor.Schema, credentials)
+		if len(filtered) == 0 {
+			return nil, errors.New("credentials do not satisfy the requirements")
```
Comment (llorllale) [inline pkg/doc/presexch/definition.go:188]:
```suggestion
			return nil, fmt.Errorf("credentials do not satisfy requirements with schema %s", descriptor.Schema)
```

File: pkg/doc/presexch/definition.go [hunk 3]
```diff
@@ -166,3 +168,90 @@ func (pd *PresentationDefinition) ValidateSchema() error {
...
+	}
+
+	vp := &verifiable.Presentation{}
+
+	err := vp.SetCredentials(credentialsToInterface(merge(result))...)
+	if err != nil {
+		return nil, err
+	}
+
+	return vp, nil
```
Comment (llorllale) [inline pkg/doc/presexch/definition.go:201]: This VP is left without context, schema, etc. You'd have to manually set them on line 194.

(we could really use builder APIs for VPs....)

File: pkg/doc/presexch/definition_test.go [hunk 4]
```diff
@@ -39,6 +41,286 @@ func TestPresentationDefinition_IsValid(t *testing.T) {
 	})
 }
 
+func TestPresentationDefinition_CreateVP(t *testing.T) {
```
Comment (llorllale) [inline pkg/doc/presexch/definition_test.go:44]: We should include tests that check whether the output VP is formed correctly (should pass a parsing check). This is related to by comment [here](https://github.com/hyperledger/aries-framework-go/pull/2407/files#r545164335).

File: pkg/doc/presexch/definition.go (hunk 3 above)
Comment (soluchok) [reply pkg/doc/presexch/definition.go:201]: VP will be improved. The final result must be a valid VP. We do not have builder APIs :(

File: pkg/doc/presexch/definition_test.go (hunk 4 above)
Comment (soluchok) [reply pkg/doc/presexch/definition_test.go:44]: I will add this check

File: pkg/doc/presexch/definition.go (hunk 3 above)
Comment (sudeshrshetty) [reply pkg/doc/presexch/definition.go:201]: I don't `presentation_submission` in Verifiable Presentation you generated, is it going to be added in future?

Comment (soluchok) [reply pkg/doc/presexch/definition.go:201]: @sudeshrshetty yes, it will be added soon

**Review by sudeshrshetty — APPROVED** [review]

File: pkg/doc/presexch/definition_test.go (hunk 4 above)
Comment (soluchok) [reply pkg/doc/presexch/definition_test.go:44]: @llorllale added `context` and `type`. Also added check in tests whether the VP is parsable.

**Review by llorllale — APPROVED** [review]

**Pull request merged**